PurchasesCoreSwift/Attribution/AttributionFetcher.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Created by Andrés Boedo on 4/8/21.
+// Copyright (c) 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+#if os(iOS) || os(tvOS)
+import UIKit
+#elseif os(watchOS)
+import WatchKit
+#endif
+
+enum AttributionFetcherError: Error {
+
+    case identifierForAdvertiserUnavailableForPlatform
+    case identifierForAdvertiserFrameworksUnavailable
+
+}
+
+@objc(RCAttributionFetcher) public class AttributionFetcher: NSObject {
+    private let attributionFactory: AttributionTypeFactory
+    private let systemInfo: SystemInfo
+
+    #if os(watchOS) || os(macOS) || targetEnvironment(macCatalyst)
+    private let appTrackingTransparencyRequired = false
+    #else
+    private let appTrackingTransparencyRequired = true
+    #endif
+
+
+    @objc public init(attributionFactory: AttributionTypeFactory, systemInfo: SystemInfo) {
+        self.attributionFactory = attributionFactory
+        self.systemInfo = systemInfo
+    }
+
+    @objc public var identifierForVendor: String? {
+        // Should match available platforms in
+        // https://developer.apple.com/documentation/uikit/uidevice?language=swift
+        // https://developer.apple.com/documentation/watchkit/wkinterfacedevice?language=swift
+        #if os(iOS) || os(tvOS)
+            UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            WKInterfaceDevice.current().identifierForVendor?.uuidString
```

Comment:
in obj-c we were actually not handling the watchOS case, so this would just always be `nil` for watchOS. 
I think it's just a result of the fact that watchOS support was added much later, we probably just forgot to add it here.

Reply:
Oops